### PR TITLE
hiddify: init at 1.5.2

### LINKED ISSUE #285375: Package request: hiddify-next
**Project description**

A multi-platform proxy client based on Sing-box universal proxy tool-chain. Hiddify Next offers a wide range of capabilities, like automatic node selection, TUN mode, remote profiles etc. Hiddify Next is ad-free and open-source. With support for a wide range of protocols, it provides a secure and private way for accessing free internet.

Note, From the [Archwiki](https://wiki.archlinux.org/title/V2Ray): V2ray has become mostly "unmaintainable" and has to be replaced by Sing-Box/shadowsocks-go...

**Metadata**

* homepage URL: https://hiddify.com/
* source URL:  https://github.com/hiddify/hiddify-next
* license: Attribution-NonCommercial-ShareAlike 4.0 InternationalJ
* platforms: Windows, Mac, Linux, Android

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by BeiyanYunyi**
## Description of changes

- Init hiddify at 1.5.2
- Adds source builder for `sqlite3_flutter_libs` flutter dependency as needed
- Add `BeiyanYunyi` to `maintainer-list.nix`

Closed #303085 as duplicate (that pr also ships with an older version, and uses AppImage while this compiles from source)

Fix https://github.com/NixOS/nixpkgs/issues/285375, related to https://github.com/NixOS/nixpkgs/issues/287770

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  extraWrapProgramArgs = ''
+    --prefix LD_LIBRARY_PATH : "$out/app/lib"
+  '';
+
+  gitHashes = {
+    circle_flags = "sha256-dqORH4yj0jU8r9hP9NTjrlEO0ReHt4wds7BhgRPq57g=";
+  };
+
+  meta = with lib; {
+    description = "Multi-platform auto-proxy client, supporting Sing-box, X-ray, TUIC, Hysteria, Reality, Trojan, SSH etc. It’s an open-source, secure and ad-free. ";
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:85]:
```suggestion
    description = "Multi-platform auto-proxy client";
    longDescription = ''
      Multi-platform auto-proxy client, supporting Sing-box, X-ray, TUIC, Hysteria,
      Reality, Trojan, SSH etc. It’s an open-source, secure and ad-free.
    '';
```

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  extraWrapProgramArgs = ''
+    --prefix LD_LIBRARY_PATH : "$out/app/lib"
+  '';
+
+  gitHashes = {
+    circle_flags = "sha256-dqORH4yj0jU8r9hP9NTjrlEO0ReHt4wds7BhgRPq57g=";
+  };
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:84]: Better to avoid `with lib`

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    description = "Multi-platform auto-proxy client, supporting Sing-box, X-ray, TUIC, Hysteria, Reality, Trojan, SSH etc. It’s an open-source, secure and ad-free. ";
+    homepage = "https://github.com/hiddify/hiddify-next/";
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:86]:
```suggestion
    homepage = "https://hiddify.com";
```

File: pkgs/by-name/hi/hiddify-next/package.nix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+, pkg-config
+, flutter
+, buildGoModule
+, libayatana-appindicator
+}:
+let
+  pname = "hiddify-next";
+  version = "1.5.2";
+
+  hiddify = fetchFromGitHub {
```
Comment (Aleksanaa) [inline pkgs/by-name/hi/hiddify-next/package.nix:12]: One may want to make these packages overridable. There are some solutions:

1. use `callPackage` to load from another file, as a default argument `{ ..., hiddify? callPackage ./hiddify.nix {}, ... }: flutter... { ... }`, and expose them in `passthru` of the main package
2. use `callPackages`, and write a set of packages, like https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/language-servers/nixd/default.nix
3. just split them into different by-name directories, and inherit `version` and `src`.

**Review by Aleksanaa — COMMENTED** [review]
Others look fine. Please reword `hiddify: init at 1.5.2` to `hiddify-next: init at 1.5.2`

**Comment by TanvirOnGH** [comment]
### Update
Latest Release: [v2.0.5](https://github.com/hiddify/hiddify-next/releases/tag/v2.0.5)
Latest Pre-release: [v2.5.7](https://github.com/hiddify/hiddify-next/releases/tag/v2.5.7)

**Comment by BeiyanYunyi** [comment]
Close due to personal conflict with a main contributor of sing-box.

**Pull request closed without merging**